designdocs/provider_contexts.md

```diff
@@ -0,0 +1,231 @@
+To align with the data security model of company-to-user in the world today, permissions, and thus Provider-level delegations continue to reflect the reality best.
+
+Finally, this structure does not prevent the setup of a 1:1 relationship between Application and Provider.
+If a company wishes, they can create more than one Provider for the purposes of separation.
+This may be desirable if there is a subsidiary company or other complex structures.
+
+#### User Trust in Application Presentation
+
+Users need to know what application they are logging into and trust that the displayed information is not a phishing attack.
+Users have a reasonable expectation that the chain and wallet provide a level of protection against phishing attacks from Providers and Applications.
+
+#### Easy Wallet Integration of Data
+
+Wallets displaying login request information need to be able to easily obtain and verify the information they want to show the user.
+
+## Proposal
+
+Frequency should provide a safeguarded system of setting and updating Provider and Provider Application Contexts for wallets to display to users.
+
+- [Storage and Structure of the Data](#data)
+- [Mainnet Approval Flow](#governance)
+- [Example of Wallet Usage via SIWF](#siwf)
+- [Provider Dashboard Steps](#dashboard)
+
+### **Storage and Structure of the Data** <a id='data'></a>
+
+Currently, the `ProviderRegistryEntry` has only a `name` property, and no way to update said name.
+
+Applications have two different pieces of default and (potentially) internationalized data:
+- name
+- logo/image
+
+Limits
+1. Provide a space to internationalize up to 150 localizations (Windows and macOS each have < 150).
+2. Image MUST BE encoded (`png`)[https://www.w3.org/TR/png-3/] that is `250x100` (support for future sizes and light/dark should be considered).
+
+The data structure must support both of these and internationalization.
+
+Proposed are the following changes:
+
+1. `ProviderRegistryEntry` struct be renamed `ApplicationRegistryEntry`
+2. `ApplicationRegistryEntry` have the following properties:
+    ```rust
+    pub struct ApplicationRegistryEntry<T, U>
+    where
+        T: Get<u32>,
+        U: Get<u32>,
+    {
+        pub default_name: BoundedVec<u8, T>,
+        pub localized_names: BTreeMap<Vec<u8>, BoundedVec<u8, T>>,
+        pub default_logo_250_100_png_bytes: BoundedVec<u8, U>,
+        pub localized_logo_250_100_png_bytes: BTreeMap<Vec<u8>, BoundedVec<u8, U>>,
+    }
+    ```
+3. The `ProviderToRegistryEntry` be updated to use `ProviderToApplicationRegistryEntry`
+    ```rust
+    #[pallet::storage]
+	pub type ProviderToRegistryEntry<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ProviderId,
+		ProviderToApplicationRegistryEntry<T::MaxProviderNameSize, T::MaxProviderLogo250X100Size>,
```

Comment:
@JoeCap08055 This was a typo. It should ALSO be a `ApplicationRegistryEntry` (or an alias'd version of it)

```suggestion
3. The `ProviderToRegistryEntry` be updated to use `ApplicationRegistryEntry` so that the Provider has a "default" `ApplicationRegistryEntry`.
    ```rust
    #[pallet::storage]
	pub type ProviderToRegistryEntry<T: Config> = StorageMap<
		_,
		Twox64Concat,
		ProviderId,
		ApplicationRegistryEntry<T::MaxProviderNameSize, T::MaxProviderLogo250X100Size>,
```